datasource-decorator-spring-boot-autoconfigure/src/main/java/com/github/gavlyukovskiy/micrometer/DataSourceMetricsHolder.java

```diff
@@ -91,7 +93,6 @@ void afterAcquireConnection(Connection connection, long timeToAcquireConnection,
             connectionAcquireTimestamp.put(connection, System.currentTimeMillis());
```

Comment:
You definitely want to use `System.nanoTime()` here rather than `System.currentTimeMillis()`. Nanos are guaranteed to be monotonically increasing (and have greater precision for timings anyway). Millis can go back in time or jump forward at unpredictable intervals if someone sets the time. On many systems ntpd will be constantly keeping the time synced up with an authoritative source. This could result in a negative timing!

Reply:
I can use nanos only with P6Spy, unfortunately datasource-proxy uses millis. Although I can calculate time by myself.